src/sentry_core.c

```diff
@@ -211,6 +246,8 @@ sentry_init(sentry_options_t *options)
         }
         sentry_value_freeze(scope->client_sdk);
         initialize_propagation_context(&scope->propagation_context);
+        // TODO pass in scope, or make it call SENTRY_WITH_SCOPE?
+        set_dynamic_sampling_context(scope);
```

Comment:
I'm not 100% sure what function signature makes more sense here; for now I felt like passing in the scope was most logical.

Reply:
Good question! While all our mutexes are recursive, we shouldn't overdo them. If you have a static helper that is invoked from a function that has already retrieved shared memory, please always pass references to it down. We should actively seek and eliminate unnecessary shared memory retrievals by argument passing.

The primary reason we allow/need recursive mutexes is that we internally reuse API entry points that require access to global SDK state.